src/nupic/engine/NetworkFactory.cpp

```diff
@@ -0,0 +1,257 @@
+      }
+
+      if ((*link).size() != 6)
+      { 
+        NTA_THROW << "Invalid network structure file -- bad link (wrong size)";
+      }
+
+      // 1. type
+      node = (*link).FindValue("type");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have a type";
+      }
+      std::string linkType;
+      *node >> linkType;
+
+      // 2. params
+      node = (*link).FindValue("params");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have params";
+      }
+
+      std::string params;
+      *node >> params;
+
+      // 3. srcRegion (name)
+      node = (*link).FindValue("srcRegion");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have a srcRegion";
+      }
+
+      std::string srcRegionName;
+      *node >> srcRegionName;
+
+      // 4. srcOutput
+      node = (*link).FindValue("srcOutput");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have a srcOutput";
+      }
+
+      std::string srcOutputName;
+      *node >> srcOutputName;
+
+      // 5. destRegion
+      node = (*link).FindValue("destRegion");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have a destRegion";
+      }
+
+      std::string destRegionName;
+      *node >> destRegionName;
+
+      // 6. destInput
+      node = (*link).FindValue("destInput");
+      if (node == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link does not have a destInput";
+      }
+
+      std::string destInputName;
+      *node >> destInputName;
+
+      if (!regionList.contains(srcRegionName))
+      {
+        NTA_THROW << "Invalid network structure file -- link specifies source region '" << srcRegionName << "' but no such region exists";
+      }
+
+      Region* srcRegion = regionList.getByName(srcRegionName);
+
+      if (!regionList.contains(destRegionName))
+      {
+        NTA_THROW << "Invalid network structure file -- link specifies destination region '" << destRegionName << "' but no such region exists";
+      }
+
+      Region* destRegion = regionList.getByName(destRegionName);
+
+      Output* srcOutput = srcRegion->getOutput(srcOutputName);
+      if (srcOutput == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link specifies source output '" << srcOutputName << "' but no such name exists";
+      }
+
+      Input* destInput = destRegion->getInput(destInputName);
+      if (destInput == nullptr)
+      {
+        NTA_THROW << "Invalid network structure file -- link specifies destination input '" << destInputName << "' but no such name exists";
+      }
+
+      // Create the link itself
+
+      n.link(srcRegionName, destRegionName, linkType, params, srcOutputName, destInputName);
+    }
+    return(n);
```

Comment:
nit: `return` is a keyword, not a function, so it is more common to write this as `return n;`